Intersect.Client/Entities/Entity.cs

```diff
@@ -1386,6 +1386,147 @@ public virtual void DrawName(Color textColor, Color borderColor = null, Color ba
             );
         }
 
+        // Tags
+        public virtual void DrawTag(NpcTagPos npcTagPos, PlayerTagPos playerTagPos)
+        {
+            // Variables
+            string tagFileName = "";
+            string entityName = this.Name;
+            bool customNpcTag = Options.Npc.CustomTagIcons.Contains(entityName);
+            bool customPlayerTag = Options.Player.CustomTagIcons.Contains(entityName);
+            bool customNpcTagOnly = Options.Npc.ShowCustomTagsOnly;
+            var nameSize = Graphics.Renderer.MeasureText(entityName, Graphics.EntityNameFont, 1);
+            var nameCentHorPos = (int)Math.Ceiling(GetCenterPos().X);
+            var nameVertPos = GetLabelLocation(LabelType.Name);
+            float x;
+            float y;
+            // Feature check
+            if (!Options.Npc.ShowTags || !Options.Player.ShowTags)
```

Comment:
The `!Options.Player.ShowTags` check should be done in an override method in the Player class. I mention this further down in the review.

Reply:
👍🏽 
```
            // Feature and Entity check
            if (!Options.Player.ShowTags || !customPlayerTag || !(this is Player))
            {
                return;
            }
```